lib/server.go

```diff
@@ -2253,14 +2259,16 @@ func (srv *Server) _handleFastHostStuffTimeout(event *consensus.ConsensusEvent)
 	// 4. Broadcast the timeout msg to the network
 }
 
-func (srv *Server) _handleFastHostStuffConsensusEvent(event *consensus.ConsensusEvent) {
+func (srv *Server) _handleFastHostStuffConsensusEvent(event *consensus.FastHotStuffEvent) {
 	switch event.EventType {
-	case consensus.ConsensusEventTypeBlockProposal:
-		srv._handleFastHostStuffBlockProposal(event)
-	case consensus.ConsensusEventTypeVote:
+	case consensus.FastHotStuffEventTypeVote:
 		srv._handleFastHostStuffVote(event)
-	case consensus.ConsensusEventTypeTimeout:
+	case consensus.FastHotStuffEventTypeTimeout:
 		srv._handleFastHostStuffTimeout(event)
+	case consensus.FastHotStuffEventTypeConstructVoteQC:
+		srv._handleFastHostStuffBlockProposal(event)
+	case consensus.FastHotStuffEventTypeConstructTimeoutQC:
+		srv._handleFastHostStuffEmptyTimeoutBlockProposal(event)
 	}
 }
```

Comment:
I like this structure a lot. You've managed to isolate most of the fast-hotstuff complexity in a separate lib, where it can be rigorously tested, and this interface for it in server.go is super clean. Basically server.go is the "glue" between the consensus logic and the actual real connections to other nodes. Love it.